src/components/posts/PostDetail.js

```diff
@@ -55,36 +55,24 @@ export default (props) => {
         </Col>
         <Col className="d-flex justify-content-end">
           <p className="postDetail__category">
-            <Badge variant="info">{post.category.name}</Badge>
+            <Badge variant="info">{post.category.label}</Badge>
           </p>
         </Col>
       </Row>
 
       <Row className="justify-content-center my-4">
         <Image src={post.image} fluid />
       </Row>
+      <Row className="justify-content-center my-4">
+      <Button variant="secondary"
+      type="submit"
+      className="ml-2"
+      onClick={e=> props.history.push(`posts/${post.id}/comments`)}
```

Comment:
I think the route you specify here needs to start with a leading slash. So, `/posts/${post.id}/comments` instead of just `posts/${post.id}/comments`. Because it doesn't have the leading slash, right now when I click on it I am getting redirected to the URL of `/posts/posts/2/comments`, which is one posts too many.